tests/tools/tools-object.test.js

```diff
@@ -0,0 +1,161 @@
+const { convertTools, createToolObject } = require('../../scripts/tools/tools-object');
+const {
+  createToolFileContent,
+  createExpectedToolObject,
+  createMockData,
+  createMalformedYAML
+} = require('../helper/toolsObjectData');
+
+jest.mock('axios');
+jest.mock('../../scripts/tools/categorylist', () => ({
+  categoryList: [
+    { name: 'Category1', tag: 'Category1', description: 'Description for Category1' },
+    { name: 'Others', tag: 'Others', description: 'Other tools category' },
+  ]
+}));
+
+describe('Tools Object', () => {
+  beforeEach(() => {
+    axios.get.mockClear();
+    console.error = jest.fn();
+  });
+
+  const mockToolData = (toolContent, toolNames = ['valid-tool']) => {
+    const mockData = createMockData(toolNames.map((name) => ({ name: `.asyncapi-tool-${name}`, repoName: name })));
+    axios.get.mockResolvedValue({ data: toolContent });
+    return mockData;
+  };
+
+  it('should create a tool object with provided parameters', async () => {
+    const toolFile = createToolFileContent({
+      title: 'Test Tool',
+      description: 'Test Description',
+      hasCommercial: true,
+      additionalLinks: { docsUrl: 'https://docs.example.com' }
+    });
+
+    const expected = createExpectedToolObject({
+      title: 'Test Tool',
+      description: 'Test Description',
+      hasCommercial: true,
+      additionalLinks: { docsUrl: 'https://docs.example.com' }
+    });
+
+    const result = await createToolObject(
+      toolFile,
+      expected.links.repoUrl,
+      'Repository Description',
+      true
+    );
+
+    expect(result).toEqual(expected);
+  });
+
+  it('should convert tools data correctly', async () => {
+    const toolContent = createToolFileContent({ title: 'Valid Tool', categories: ['Category1'] });
+    const mockData = mockToolData(toolContent);
+
+    const result = await convertTools(mockData);
+
+    expect(result.Category1.toolsList).toHaveLength(1);
+    expect(result.Category1.toolsList[0].title).toBe('Valid Tool');
+  });
+
+  it('should assign tool to Others category if no matching category is found', async () => {
+    const toolContent = createToolFileContent({ title: 'Unknown Category Tool', categories: ['UnknownCategory'] });
+    const mockData = mockToolData(toolContent);
+
+    const result = await convertTools(mockData);
+
+    expect(result.Others.toolsList).toHaveLength(1);
+    expect(result.Others.toolsList[0].title).toBe('Unknown Category Tool');
+  });
+
+  it('should log errors for invalid .asyncapi-tool file', async () => {
+    const invalidContent = createToolFileContent({
+      title: 'Invalid Tool',
+      additionalFilters: { invalidField: true }
+    });
+    const mockData = mockToolData(invalidContent);
+
+    await convertTools(mockData);
+
+    expect(console.error).toHaveBeenCalledWith(expect.stringContaining('Script is not failing'));
+    expect(console.error).toHaveBeenCalledWith(expect.stringContaining('Invalid .asyncapi-tool file'));
+  });
```

Comment:
@vishvamsinh28 Can you please apply this suggestion?

Reply:
@akshatnema This will fail the test because the script logs errors in multiple lines, whereas the suggestion expects a single log entry containing all error details as a JSON string.